src/Context/Settings/index.jsx

```diff
@@ -1,53 +1,50 @@
-import React, {useState, useEffect} from 'react';
-
-
-// create a context object.......
+import React, { useState, useEffect } from 'react';
 
 export const SettingsContext = React.createContext();
 
-// create a provider component
-function SettingsProvider({ children }){
-    const [ displayCount, setDisplayCount ] = useState(3);
-    const [ showCompleted, setShowCompleted ] = useState(false);
-    const [ sortField, setSortField ] = useState('difficulty');
+function SettingsProvider({ children }) {
+  const [displayCount, setDisplayCount] = useState(3);
+  const [showCompleted, setShowCompleted] = useState(false);
+  const [sortField, setSortField] = useState('difficulty');
+  const [itemsToShow, setItemsToShow] = useState(5);
 
-  // Read settings from Local Storage on component mount
   useEffect(() => {
     const storedSettings = localStorage.getItem('settings');
     if (storedSettings) {
-      const { displayCount, showCompleted, sortField } = JSON.parse(storedSettings);
+      const { displayCount, showCompleted, sortField, itemsToShow } = JSON.parse(storedSettings);
       setDisplayCount(displayCount);
       setShowCompleted(showCompleted);
       setSortField(sortField);
+      setItemsToShow(itemsToShow);
     }
   }, []);
 
-    // Update Local Storage when settings change
-    useEffect(() => {
-        const settings = {
-          displayCount,
-          showCompleted,
-          sortField,
-        };
-        localStorage.setItem('settings', JSON.stringify(settings));
-      }, [displayCount, showCompleted, sortField]);
-
-    // this object contains the state and any functions we need to share
-    const state = {
-        displayCount,
-        showCompleted,
-        sortField,
-        setDisplayCount,
-        setShowCompleted,
-        setSortField,
-    }
-    
-    return (
-        <SettingsContext.Provider value={state}>
-            {children}
-        </SettingsContext.Provider>
-
-    )
+  useEffect(() => {
+    const settings = {
+      displayCount,
+      showCompleted,
+      sortField,
+      itemsToShow,
+    };
+    localStorage.setItem('settings', JSON.stringify(settings));
+  }, [displayCount, showCompleted, sortField, itemsToShow]);
```

Comment:
You still have the same bug as yesterday's lab, resetting your settings to default on page load before you can pull them from localStorage. Another way to deal with this would be to create a new state:
```js
const [firstRender, setFirstRender] = useState(true)

useEffect(() => {
    if (firstRender) {
        setFirstRender(false)
    } else {
       {* set localStorage *}
    }
}, [settings])
```

Reply:
This has been added.